frontend/src/components/FindMatch.js

```diff
@@ -134,6 +110,36 @@ const FindMatch = () => {
     )
   }
 
+  matchingSocket.on('found-connection', (username, difficulty) => {
+    const room = {
+      room_id: username,
+      id1: username,
+      id2: user.username,
+      id1_present: true,
+      id2_present: true,
+      difficulty: difficulty.toLowerCase(),
+      datetime: new Date(),
+    }
+    setRoom(room)
+  })
+
+  useEffect(() => {
+    if (room) {
+      try {
+        createRoomService(room)
+      } catch (e) {
+        console.log(e)
+      } finally {
+        navigate(collabUrl, {
+          state: {
+            room: room.room_id,
+            difficulty: room.difficulty,
+          },
+        })
+      }
```

Comment:
Just wondering, why is the page navigation in a `finally` block? Cos I was thinking if the room fails to be created, then we shouldn't let the user proceed right?

Reply:
eh i think you are right! previous code base i forget to account for that!